File: WasteVisionWebFE/src/data/mappers/log-mapper.ts
import { ObjectPrediction } from "@/core/domain/ObjectPrediction";
import {
  LogDTO,
} from "../dto/log-dto";
import { Log } from "@/core/domain/log";

export class LogMapper {
  /**
   * Maps an LogDTO to a domain Log
   */
  public static toDomain(dto: LogDTO): Log {
    return {
      type: dto.type,
      timestamp: dto.timestamp,
      description: dto.description,
    };
  }

  /**
   * Maps a domain Log to a LogDTO
   */
  public static toDTO(domain: Log): LogDTO {
    return {
      type: domain.type,
      timestamp: domain.timestamp,
      description: domain.description,
    };
  }

  /**
   * Maps an array of LogDTOs to domain Logs
   */
  public static toDomainList(dtos: LogDTO[]): Log[] {
    return dtos.map((dto) => this.toDomain(dto));
  }

  /**
   * Maps an array of domain Logs to LogDTOs
   */
  public static toDTOList(domains: Log[]): LogDTO[] {
    return domains.map((domain) => this.toDTO(domain));
  }

}

File: WasteVisionWebBE/Domain/Logging/LogMapper.cs
using System.Collections.Generic;

namespace DDDSample1.Domain.Logging
{
public class LogMapper
    {
        public static LogDto toDto(LogEntry logEntry)
        {
            return new LogDto
            {
                Type = logEntry.Type.ToString(),
                Timestamp = logEntry.Timestamp,
                Description = logEntry.Description
            };
        }

        public static LogEntry toDomain(LogDto logDto)
        {
            return new LogEntry
            {
                Type = LogType.FromString(logDto.Type),
                Timestamp = logDto.Timestamp,
                Description = logDto.Description
            };
        }

        public static IEnumerable<LogDto> toDtoList(IEnumerable<LogEntry> logEntries)
        {
            List<LogDto> logDtos = new List<LogDto>();

            foreach (LogEntry logEntry in logEntries)
            {
                logDtos.Add(toDto(logEntry));
            }

            return logDtos;
        }

        public static IEnumerable<LogEntry> toDomainList(IEnumerable<LogDto> logDtos)
        {
            List<LogEntry> logEntries = new List<LogEntry>();

            foreach (LogDto logDto in logDtos)
            {
                logEntries.Add(toDomain(logDto));
            }

            return logEntries;
        }
    }
}